### PR TITLE
[RISC-V] Fix assertion for auipc and lui

### PR DESCRIPTION
**Pull request by clamp03**
Fix https://github.com/dotnet/runtime/pull/99278#issuecomment-2033667643 issue

Part of https://github.com/dotnet/runtime/issues/84834
cc @dotnet/samsung

### CODE REVIEW

**Comment by shushanhf** [comment]
Thanks very much~

**Comment by clamp03** [comment]
risc-v coreclr test is done (QEMU + Checked) and there is no additional error.

**Review by sirntar — APPROVED** [review]

File: src/coreclr/jit/emitriscv64.cpp [hunk 1]
```diff
@@ -1253,8 +1253,9 @@ void emitter::emitLoadImmediate(emitAttr size, regNumber reg, ssize_t imm)
     // Since ADDIW use sign extension fo immediate
     // we have to adjust higher 19 bit loaded by LUI
     // for case when low part is bigger than 0x800.
-    UINT32 high19 = (high31 + 0x800) >> 12;
+    INT32 high19 = ((int32_t)(high31 + 0x800)) >> 12;
 
+    assert(isValidSimm20(high19));
```
Comment (Bajtazar) [inline src/coreclr/jit/emitriscv64.cpp:1258]: NIT: This assert is redundant since the immediate value will also be checked in the `emitIns_R_I` function

Comment (clamp03) [reply src/coreclr/jit/emitriscv64.cpp]: Thank you! Removed.

**Review by Bajtazar — APPROVED** [review]

**Comment by clamp03** [comment]
@jakobbotsch Could you review this PR?

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**